File: browsergameMapEditor/models/MapInfo.py
from typing import List

from models.JSONAble import JSONAble
from models.MapTile import MapTile


class MapInfo(JSONAble):
    def __init__(self, mapWidth: int, mapHeight: int, region: int, tiles: List[MapTile]):
        super().__init__()
        self.MapWidth = mapWidth
        self.MapHeight = mapHeight
        self.Region = region
        self.Tiles = tiles

File: BrowserGameServer/BrowserGameServer/GameServer/Map/MapInfo.cs
namespace BrowserGameServer.GameServer.Map;

public class MapInfo
{
    public int MapWidth { get; init; }
    public int MapHeight { get; init; }
    
    public int Region { get; init; }
    public List<MapTile> Tiles { get; init; }

    public MapInfo(int mapWidth, int mapHeight, int region, List<MapTile> tiles)
    {
        MapWidth = mapWidth;
        MapHeight = mapHeight;
        Tiles = tiles;
        Region = region;
    }
}